accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,502 @@
+  with duplicated logic, but it results in really ugly MSBuild, since it's not a
+  very expressive programming language. It also creates the potential for drift
+  between the two definitions. That said, exposing it directly would create a
+  new tight coupling between MSBuild and NuGet that hasn't historically existed.
+  It would probably require a direct dependency and update flow plus coherency
+  requirements on both .NET Core SDK and VS insertions. If the logic were in a
+  separate package (we've talked about pushing it down to the framework at
+  various times) it'd be great to just expose that. With the logic in NuGet,
+  it's reasonable either way, just different tradeoffs. I'm amenable to exposing
+  a property function, but maybe we should go down the road of not doing it at
+  first.
+
+We're going to map the new entries as follows:
+
+| Framework      | Identifier    | Version | Profile |
+|----------------|---------------|---------|---------|
+| net48          | .NETFramework | 4.8     |         |
+| net5.0         | .NETCoreApp   | 5.0     |         |
+| net5.0-android | .NETCoreApp   | 5.0     | android |
+| net5.0-ios     | .NETCoreApp   | 5.0     | ios     |
+| net5.0-win     | .NETCoreApp   | 5.0     | win     |
+
+***Open Issue**. Please note that `net5.0`+ will map the TFI to `.NETCoreApp`.
+We need to announce this change so that package authors with custom .props and
+.targets are prepared. Link to DavKean’s doc on how to do it.*
+
+***Open Issue**. We should try to keep the TFI out of the .nuspec file. It seems
+NuGet uses the long form `.NETFramework,Version=4.5` in the dependency groups.
+We may want to change NuGet to allow the short form there as well and update our
+packaging tools to re-write to short name on pack.
+
+Specifically:
+
+* **We'll continue to use .NETCoreApp as the TFI**. This reduces the number of
+  pieces and build logic that needs to change in .NET SDK.
+
+* **net4x and earlier will continue to use .NETFramework as the TFI**. This
+  means that net4 and net5 aren't considered compatible by default, but the
+  compatibility will continue to be handled by the AssetTargetFallback in the
+  SDK/NuGet restore.
+
+* **We'll use the existing profile concept to encode the OS selection.** It's
+  worth noting that profiles today make things smaller whereas this use makes
+  things larger. We believe this can be handled in the NuGet layer by using this
+  rule for net5 and up. While that's not ideal, it reduces the number of
+  concepts we need to add.
+
+### TFMs are a closed set
+
+One question is whether third parties can extend the TFM space (i.e. the part
+after the dash) without having to rely on changes in NuGet/MSBuild. Due to the
+expansion of short name into TFI & TFM this would be non-trivial.
+
+We may open this up in the future, but for now the consensus was that we'd
+prefer to have the specific list of names that are expanded by NuGet.
+
+### What about .NET 10?
```

Comment:
Here is an idea:

What if you call the TFMs after the convergence `dotnet` instead of just `net`. So it would be `dotnet5.0`, `dotnet5.0-win`, `dotnet10.0` and so on. In my view this would generally solve the confusion with all the existing TFMs (e.g. `net10`,`net20`,`net48`, netstandard2.0, `netcoreapp3.1`, `netcore45`).  `dotnet10.0` cannot be confused with the existing `net10`.

And `net5.0` is not just 0.2 better than `net48`! 

The convergence marks an important turning point in the history of the thing that once was called ".NET" and from now on shall only be referred to as the one and only "dotnet".  

https://twitter.com/bitbonk/status/1228597458764214272

Reply:
https://docs.microsoft.com/en-us/nuget/reference/target-frameworks#deprecated-frameworks
`dotnet50` to `dotnet56` also already have meanings. Is `dotnet50` 10 times better than `dotnet5`?

Reply:
What did these *deprecated* TFMs actually mean? Have they ever been actually used? Does a nupkg with such a TFM exist on nuget.org?

Reply:
We're not going to use `dotnet` because `net` has way more prior art and will feel more natural. The issue with dots exists everywhere; it's a minor issue and NuGet can start to issue warnings when dots are missing, which is tracked by https://github.com/NuGet/Home/issues/9215.